### ISSUE TITLE
No module named ipykernel_launcher

### ISSUE THREAD

**Issue by Jayfeath3r** [jupyter/notebook#5738]
I just install the notebook on my Python 3.8.5 by "pip install notebook" however when I want to run code it always says " connection failed" 

Here is my information on CMD
D:\>jupyter notebook
[I 18:50:10.057 NotebookApp] The port 8888 is already in use, trying another port.
[I 18:50:10.753 NotebookApp] Serving notebooks from local directory: D:\
[I 18:50:10.754 NotebookApp] Jupyter Notebook 6.1.4 is running at:
[I 18:50:10.755 NotebookApp] http://localhost:8889/?token=8ad122ee40323a890964325f6ca2e7d6928f4e0b9710cc19
[I 18:50:10.758 NotebookApp]  or http://127.0.0.1:8889/?token=8ad122ee40323a890964325f6ca2e7d6928f4e0b9710cc19
[I 18:50:10.759 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 18:50:11.053 NotebookApp]

    To access the notebook, open this file in a browser:
        file:///C:/Users/Owner/AppData/Roaming/jupyter/runtime/nbserver-13556-open.html
    Or copy and paste one of these URLs:
        http://localhost:8889/?token=8ad122ee40323a890964325f6ca2e7d6928f4e0b9710cc19
     or http://127.0.0.1:8889/?token=8ad122ee40323a890964325f6ca2e7d6928f4e0b9710cc19
[I 18:50:24.358 NotebookApp] Kernel started: 7e1c8010-786d-4cc3-a72e-e0f08daa6cbc, name: python3
d:\python.exe: No module named ipykernel_launcher
[I 18:50:27.284 NotebookApp] KernelRestarter: restarting kernel (1/5), new random ports
d:\python.exe: No module named ipykernel_launcher
[I 18:50:30.309 NotebookApp] KernelRestarter: restarting kernel (2/5), new random ports
d:\python.exe: No module named ipykernel_launcher
[I 18:50:33.321 NotebookApp] KernelRestarter: restarting kernel (3/5), new random ports
d:\python.exe: No module named ipykernel_launcher
[I 18:50:36.343 NotebookApp] KernelRestarter: restarting kernel (4/5), new random ports
d:\python.exe: No module named ipykernel_launcher
[W 18:50:39.371 NotebookApp] KernelRestarter: restart failed
[W 18:50:39.371 NotebookApp] Kernel 7e1c8010-786d-4cc3-a72e-e0f08daa6cbc died, removing from map.
[W 18:51:24.430 NotebookApp] Timeout waiting for kernel_info reply from 7e1c8010-786d-4cc3-a72e-e0f08daa6cbc
[E 18:51:24.434 NotebookApp] Error opening stream: HTTP 404: Not Found (Kernel does not exist: 7e1c8010-786d-4cc3-a72e-e0f08daa6cbc)
[W 18:51:25.916 NotebookApp] 404 GET /api/kernels/7e1c8010-786d-4cc3-a72e-e0f08daa6cbc/channels?session_id=2ccd41e52bfe41e698f69921bd5b39eb (::1): Kernel does not exist: 7e1c8010-786d-4cc3-a72e-e0f08daa6cbc
[W 18:51:25.928 NotebookApp] 404 GET /api/kernels/7e1c8010-786d-4cc3-a72e-e0f08daa6cbc/channels?session_id=2ccd41e52bfe41e698f69921bd5b39eb (::1) 13.96ms referer=None
[W 18:51:28.917 NotebookApp] Replacing stale connection: 7e1c8010-786d-4cc3-a72e-e0f08daa6cbc:2ccd41e52bfe41e698f69921bd5b39eb

**Comment by web-sys1** [comment]
@Jayfeath3r Jayfeath3r Same problem, while have been trying to do in software application under pyinstaller.

**Comment by kevin-bates** [comment]
Please locate the `kernel.json` file associated with the kernel you're trying to start by running `jupyter kernelspec list` and locating the kernel spec's directory.  It strikes me as odd that you have `python.exe` at the root of your drive and perhaps it doesn't have the necessary information to locate other python modules like `ipykernel_launcher`.  Once the `kernel.json` has been located, focus on the `argv` stanza.   

[This section of the docs](https://jupyter-notebook.readthedocs.io/en/latest/troubleshooting.html#what-to-do-when-things-go-wrong) may provide some additional tips.

**Comment by web-sys1** [comment]
@kevin-bates Yes, but there is a problem. Does **Jupyter Notebook** works after the application is built to `.exe` under Pyinstaller? Maybe, waste time. Please help me. This issue #3246 was left unresolved.

**Comment by zwarshavsky** [comment]
I have just discovered the same issue while associating ipykernel with python 3.8. When I downgraded to Python 3.7, the issue was resolved. `kernel.json` was identical in both the 3.7 and 3.8 environments.

**Comment by kevin-bates** [comment]
This sounds like `ipykernel` (which provides `ipykernel_launcher`) wasn't properly installed.  Try uninstalling/installing `ipykernel` in your 3.8 env.
